### PR TITLE
[Module] Topup

### PR DESCRIPTION
**Pull request by arnaudbore**
## Describe your changes

Add Topup module - Combinaison of prepare_dwi_for_topup and Topup from tractoflow DSL1

## Checklist before requesting a review

- Create the tool:
  - [x] Edit `./modules/nf-scil/<category>/<tool>/main.nf`
  - [x] Edit `./modules/nf-scil/<category>/<tool>/meta.yml`
- Generate the tests:
  - [x] Edit `./tests/modules/nf-scil/<category>/<tool>/main.nf`
  - [x] Edit `./tests/modules/nf-scil/<category>/<tool>/nextflow.config`
  - [x] Add test data locally for tests with the fork repository
  - [x] Generate the test infrastructure and _md5sum_ for all outputs
- Ensure the syntax is correct :
  - [x] Check indentation abides with the rest of the library (don't hesitate to correct others !)
  - [x] Lint everything. Ensure your variables have good names.

### CODE REVIEW

**Review by ThoumyreStanislas — COMMENTED** [review]
LGTM! Well done!

**Comment by AlexVCaron** [comment]
Test package is up to date !

File: modules/nf-scil/preproc/topup/main.nf [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+    task.ext.when == null || task.ext.when
+
+    script:
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    def prefix_topup = task.ext.prefix_topup ? task.ext.prefix_topup : ""
+    def config_topup = config_topup ?: task.ext.default_config_topup
+    def encoding = task.ext.encoding ? task.ext.encoding : ""
+    def readout = task.ext.readout ? task.ext.readout : ""
+    def b0_thr_extract_b0 = task.ext.b0_thr_extract_b0 ? task.ext.b0_thr_extract_b0 : ""
```
Comment (AlexVCaron) [inline modules/nf-scil/preproc/topup/main.nf:33]: For your knowledge, and if you want to do it here, there is a shorter way to write the ternary operator when the quantities around the `?` are the same, e.g. :

```
    (task.ext.prefix_topup ? task.ext.prefix_topup : "") === (task.ext.prefix_topup ?: "")
```

Just a bit of nice groovy flavour :stuck_out_tongue:

File: modules/nf-scil/preproc/topup/main.nf [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        --config $config_topup\
+        --encoding_direction $encoding\
+        --readout $readout --out_prefix $prefix_topup\
+        --out_script
+    sh topup.sh
+    cp corrected_b0s.nii.gz ${prefix}__corrected_b0s.nii.gz
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        scilpy: 1.6.0
```
Comment (AlexVCaron) [inline modules/nf-scil/preproc/topup/main.nf:69]:
```suggestion
        scilpy: 1.6.0
        antsRegistration: 2.4.3
        fsl: \$(flirt -version 2>&1 | sed -n 's/FLIRT version \\([0-9.]\\+\\)/\\1/p')
```

File: modules/nf-scil/preproc/topup/main.nf [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+    touch ${prefix}__corrected_b0s.nii.gz
+    touch ${prefix}__rev_b0_warped.nii.gz
+    touch ${prefix}__rev_b0_mean.nii.gz
+    touch ${prefix}__b0_mean.nii.gz
+    touch ${prefix_topup}_fieldcoef.nii.gz
+    touch ${prefix_topup}_movpar.txt
+
+    cat <<-END_VERSIONS > versions.yml
+    "${task.process}":
+        scilpy: 1.6.0
```
Comment (AlexVCaron) [inline modules/nf-scil/preproc/topup/main.nf:93]: Same comment as above

File: modules/nf-scil/preproc/topup/meta.yml [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+  - topup_fieldcoef:
+      type: file
+      description: Nifti volume - topup field to correct for distorsion
+      pattern: "*_fieldcoef.nii.gz"
+
+  - topup_movpart:
+      type: file
+      description: Text file - topup movpart
+      pattern: "*__movpar.txt"
```
Comment (AlexVCaron) [inline modules/nf-scil/preproc/topup/meta.yml:96]: Reoder this to align with the modules output order

**Review by AlexVCaron — CHANGES_REQUESTED** [review]
Very nice job, I have only minor comments, really. Once addressed, I'll just go ahead and merge ! :ship:

**Review by AlexVCaron — APPROVED** [review]

**Pull request merged**